elmslie-core/src/main/java/vivid/money/elmslie/core/store/binding/ConversionContract.kt

```diff
@@ -55,32 +50,25 @@ class ConversionContract<InitiatorEvent, InitiatorEffect, InitiatorState,
      * ```
      */
     private fun <Value, EncryptedValue> contract(
-        valueProvider: ((Value) -> Unit) -> Disposable,
+        valueProvider: Flow<Value>,
         cypher: Value.() -> EncryptedValue?,
         conversion: EncryptedValue.() -> ResponderEvent?
     ) {
         contracts += {
-            valueProvider { value ->
-                cypher(value)
-                    ?.let { encrypted -> conversion(encrypted) }
-                    ?.let(responder::accept)
+            coroutineScope.launch {
+                valueProvider.collect { value ->
+                    cypher(value)
+                        ?.let { encrypted -> conversion(encrypted) }
+                        ?.let { responder.accept(it) }
+                }
             }
         }
     }
 
-    /**
-     * Starts conversion between stores by applying contracts.
-     */
+    /** Starts conversion between stores by applying contracts. */
     fun apply() {
         check(initiator.isStarted)
         check(responder.isStarted)
-        contracts.forEach { it() }
-    }
-
-    /**
-     * Stops conversion between stores by revoking contracts.
-     */
-    fun revoke() {
```

Comment:
I understand that we are planing to deprecate contracts but let's don't delete `revoke()` method it may be a huge backward compatibility issue for some projects.